include/abb_librws/common/rw/elog.h

```diff
@@ -0,0 +1,123 @@
+#pragma once
+
+#include <string>
+#include <vector>
+#include <chrono>
+
+namespace abb ::rws ::rw ::elog
+{
+    // TODO: Replace with class that pulls domains?
+    /**
+     * @brief Elog domain enum
+     */
+    enum ElogDomain
+    {
+        COMMON = 0,
+        OPERATIONAL = 1,
+        SYSTEM = 2,
+        HARDWARE = 3,
+        PROGRAM = 4,
+        MOTION = 5,
+        IO_AND_COMMUNICATION = 7,
+        USER = 8,
+        SAFETY = 9,
+        INTERNAL = 10,
+        PROCESS = 11,
+        CONFIGURATION = 12,
+        RAPID_1 = 15,
+        RAPID_2 = 17
+    };
+
+    std::string elogDomainToString(ElogDomain domain);
+
+
+    /**
+     * @brief Types of elog messages
+     */
+    enum MessageType
+    {
+        INFORMATION = 1,
+        WARNING = 2,
+        ERROR = 3
+    };
+
+    std::string messageTypeToString(MessageType const messagType);
+
+
+    /**
+     * @brief Types of elog message arguments
+     */
+    enum ArgType
+    {
+        STRING,
+        LONG,
+        FLOAT
+    };
+
+    ArgType makeArgType(std::string const &str);
+
+    std::string argTypeToString(ArgType const &argType);
+
+    
+    /**
+     * @brief Elog message argument
+     */
+    struct ElogMessageArg
```

Comment:
What does this struct do? It is not obvious, please add a few words in the comment.